Mathlib/RingTheory/HahnSeries.lean

```diff
@@ -512,6 +512,14 @@ instance : DistribMulAction R (HahnSeries Γ V) where
     ext
     simp [mul_smul]
 
+theorem smul_order_leq {Γ} [Zero Γ] [LinearOrder Γ] (r : R) (x : HahnSeries Γ V) (h : r • x ≠ 0) :
+    x.order ≤ (r • x).order := by
+  by_cases hx : x = 0
```

Comment:
You can use `right_ne_zero_of_smul` here.

Reply:
I tried that and found that the statement of `right_ne_zero_of_smul` unnecessarily requires `[Zero R]`.  Should I change it?

Reply:
I have removed the hypothesis and moved it to `GroupTheory.GroupAction.Defs`, but I didn't know what to do with the aligns.